src/Core.js

```diff
@@ -306,6 +306,10 @@ function getLocation(url){
     }
     // #endif
     var m = url.toLowerCase().match(reURI);
+    //if the origin is non standard url, such as chrome extensions
+    if(!m) {
+        return '';
```

Comment:
I would prefer returning null, and then modifying the one (?) caller that expects null as a possible value (_getOrigin and then _window_onMessage) to handle this gracefully - actually, I think this part of the stack already does so?